jbpm-audit/src/main/java/org/jbpm/process/audit/JPAService.java

```diff
@@ -33,108 +34,134 @@
+					+ "' peristence unit is not available, no persistence strategy set for "
+					+ this.getClass().getSimpleName());
+		}
+		if (emf != null) {
+			persistenceStrategy = new StandaloneJtaStrategy(emf);
+		}
+	}
+
+	public JPAService(Environment env, PersistenceStrategyType type) {
+		persistenceStrategy = PersistenceStrategyType.getPersistenceStrategy(type, env);
+	}
+
+	public JPAService(Environment env, String peristenceUnitName) {
+		EntityManagerFactory emf = (EntityManagerFactory) env.get(EnvironmentName.ENTITY_MANAGER_FACTORY);
+		String localTransactions = (String) env.get(EnvironmentName.USE_LOCAL_TRANSACTIONS);
+
+		if (emf != null) {
+
+			if (localTransactions != null) {
+				if (localTransactions.equals("true")) {
+					persistenceStrategy = new StandaloneLocalStrategy(emf);
+				}
+			} else {
+				persistenceStrategy = new StandaloneJtaStrategy(emf);
+			}
+		} else {
+
+			if (localTransactions != null) {
+				if (localTransactions.equals("true")) {
+					persistenceStrategy = new StandaloneLocalStrategy(
+							Persistence.createEntityManagerFactory(persistenceUnitName));
+				}
+			} else {
+				persistenceStrategy = new StandaloneJtaStrategy(
+						Persistence.createEntityManagerFactory(persistenceUnitName));
+			}
+
+		}
+	}
+
+	public JPAService(EntityManagerFactory emf) {
+		persistenceStrategy = new StandaloneJtaStrategy(emf);
+	}
+
+	public JPAService(EntityManagerFactory emf, PersistenceStrategyType type) {
+		persistenceStrategy = PersistenceStrategyType.getPersistenceStrategy(type, emf);
+	}
+
+	public void setPersistenceUnitName(String persistenceUnitName) {
+		persistenceStrategy = new StandaloneJtaStrategy(Persistence.createEntityManagerFactory(persistenceUnitName));
+		this.persistenceUnitName = persistenceUnitName;
+	}
+
+	public String getPersistenceUnitName() {
+		return persistenceUnitName;
+	}
+
+	public void dispose() {
+		persistenceStrategy.dispose();
+	}
+
+	// DO NOT MAKE THIS METHOD PUBLIC!
+	// This is an internal method, and we do NOT want to expose the entity
+	// manager to users or other logic!
+	protected EntityManager getEntityManager() {
+		return persistenceStrategy.getEntityManager();
+	}
+
+	// DO NOT MAKE THIS METHOD PUBLIC!
+	// This is an internal method, and we do NOT want to expose the entity
+	// manager to users or other logic!
+	protected Object joinTransaction(EntityManager em) {
+		return persistenceStrategy.joinTransaction(em);
+	}
+
+	// DO NOT MAKE THIS METHOD PUBLIC!
+	// This is an internal method, and we do NOT want to expose the entity
+	// manager to users or other logic!
+	protected void closeEntityManager(EntityManager em, Object transaction) {
+		persistenceStrategy.leaveTransaction(em, transaction);
+	}
+
+	public <T> List<T> executeQuery(Query query, EntityManager em, Class<T> type) {
+		Object newTx = joinTransaction(em);
+		List<T> result;
+		try {
+			result = query.getResultList();
+		} finally {
+			closeEntityManager(em, newTx);
+		}
+		return result;
+	}
```

Comment:
Nit picking, but would it be possible to _not_ use tabs for indentation? We generally have 4 spaces as indent.